### ISSUE TITLE
Detect incorrect use of `|` in type annotations

### ISSUE THREAD

**Issue by Daverball** [snok/flake8-type-checking#181]
A fairly common mistake when partially wrapping/unwrapping annotations are expressions like:

```python
x: 'SomeType' | None
```

These are always invalid at runtime and generate a `TypeError` (they're fine at type checking time, since it is treated the same as `Union['SomeType', None]`)

I think it would be fairly easy to catch this mistake and add a new rule for it. Essentially any time we encounter a `ast.BinOp`/`ast.BitOr` inside a type annotation and either one of the operands is an `ast.Constant`, it should generate an error.
